### ISSUE TITLE
RecordWildCards suggested to be removed when actually it's needed

### ISSUE THREAD

**Issue by ocharles** [ndmitchell/hlint#459]
The following causes HLint to complain that RecordWildCards is unnecessary, but it's actually important:

```haskell
module A where

data T = T { foo :: Int }
```

```haskell
{-# language RecordWildCards #-}
module B where

import qualified A

bar :: A.T
bar =
  A.T { foo = 42 }
```

While it appears that I'm not using `RecordWildCards`, this extension actually subtles changes the scoping for field names in a record. Without `RecordWildCards`, you need to write:

```haskell
bar :: A.T
bar =
  A.T { A.foo = 42 }
```

**Comment by ndmitchell** [comment]
The warning is actually:

```
C:\Neil\temp>hlint B.hs
B.hs:1:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE RecordWildCards #-}
Why not remove it.
Note: you may need to add DisambiguateRecordFields
```

And it's right - in this instance you are using the `DisambiguateRecordFields` extension, and if you switch to that alone the example still works.

**Comment by ocharles** [comment]
Oh, I blindly missed that - haven't even heard of that extension! Thanks,
that should do the job. Feel free to close this.

On 21 Mar 2018 8:21 pm, "Neil Mitchell" <notifications@github.com> wrote:

The warning is actually:

C:\Neil\temp>hlint B.hs
B.hs:1:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE RecordWildCards #-}
Why not remove it.
Note: you may need to add DisambiguateRecordFields

And it's right - in this instance you are using the DisambiguateRecordFields
extension, and if you switch to that alone the example still works.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ndmitchell/hlint/issues/459#issuecomment-375082682>,
or mute
the thread

**Comment by ndmitchell** [comment]
Since I was working in that area due to #457 I improved the note to say:

```
Note: may require `{-# LANGUAGE DisambiguateRecordFields #-}` adding to the top of the file
```

Hopefully that might give a clue to the next person who stumbles into the issue.

**Comment by ocharles** [comment]
That looks great, thanks.
